test/mongooseim_loglevel_SUITE.erl

```diff
@@ -0,0 +1,196 @@
+                                          timer:seconds(5)) of
+                timeout -> ct:fail("timed out waiting for messages to reach the log file");
+                Res -> Res
+            end,
+    LinesDiff = filter_out_non_matching(After -- Before, <<"match-this">>),
+    LinesWithExpectedContents = lists:zip(LinesDiff, ExpectedContents),
+    [ ?assert('contains?'(Line, ExpectedLevel))
+      || {Line, ExpectedLevel} <- LinesWithExpectedContents ].
+
+log_at_custom_level(_Config) ->
+    %% given logging on custom log level for the helper module
+    mongoose_logs:set_global_loglevel(critical),
+    mongoose_logs:set_module_loglevel(mongooseim_loglevel_SUITE_helper, debug),
+    %% when we log from here and from the helper module
+    Before = get_log(?logfile),
+    logger:log(debug, "suite", []), % <- This is not supposed to be logged
+    mongooseim_loglevel_SUITE_helper:log(debug, "helper module", []), % <- but this should
+    %% then
+    After = case get_at_least_n_log_lines(?logfile, length(Before) + 1, timer:seconds(5)) of
+                timeout -> ct:fail("timeout waiting for messages to reach the log file");
+                Res -> Res
+            end,
+    Diff = After -- Before,
+    %% ...nothing logged from the suite reaches the log file
+    ?assertEqual([], filter_out_non_matching(Diff, <<"suite">>)),
+    %% ...logs from the helper module are found in the log file
+    [LogLine] = filter_out_non_matching(Diff, <<"helper module">>),
+    ?assert('contains?'(LogLine, <<"debug">>)).
+
+%%
+%% Helpers
+%%
+
+levels() ->
+    [
+     {-1, none},
+     {0, emergency},
+     {1, alert},
+     {2, critical},
+     {3, error},
+     {4, warning},
+     {5, notice},
+     {6, info},
+     {7, debug}
+    ].
+
+mongoose_logger_running() ->
+    File = ?logfile,
+    HandlerID = disk_log,
+    HandlerModule = logger_disk_log_h,
+    HandlerConfig = #{config => #{
+                        file => "log/mongooseim.log",
+                        type => wrap,
+                        max_no_files => 5,
+                        max_no_bytes => 2097152
+                       }
+                     },
+    ok = logger:add_handler(HandlerID, HandlerModule, HandlerConfig),
+    FileBackend = {HandlerID, File},
+    {ok, FileBackend}.
+
+levels_less_than_or_equal_to(LName) ->
+    {L, LName} = lists:keyfind(LName, 2, levels()),
+    {_, ListOfNamesLessThanOrEqual} = lists:unzip(
+              lists:takewhile(fun({LNum, _}) -> LNum =< L end, levels())
+             ),
+    ListsAtoms = lists:delete(none, ListOfNamesLessThanOrEqual),
+    lists:map(fun(El) -> atom_to_binary(El, utf8) end, ListsAtoms).
+
+'contains?'(String, Pattern) ->
+     binary:match(String, [Pattern]) /= nomatch.
+
+filter_out_non_matching(Lines, Pattern) ->
+    lists:filter(fun (L) -> 'contains?'(L, Pattern) end, Lines).
+
+get_log(LogFile) ->
+    case file:read_file(LogFile) of
+        {error, enoent} -> [];
```

Comment:
How about throwing the actual error here instead of hiding it?

Reply:
Would need to rework the logic with try-catches, we actually don't want to care for the case when the file doesn't exist yet, we might just want to retry again. Notice that this `get_log/1` function is called from functions that try it several times/

Reply:
My point here was that from this function perspective not existing file acts like it was there but empty. So this is sort of hiding the real state and imitate something else.